gesturecontrol.py

```diff
@@ -7,221 +7,259 @@
+            "mosse": cv2.TrackerMOSSE_create
+        }
+
+        # grab the appropriate object tracker using our dictionary of
+        # OpenCV object tracker objects
+        self.tracker = OPENCV_OBJECT_TRACKERS[trackerType]()
+        self.vectorX = 0.0
+        self.vectorY = 0.0
+        self.vectorZ = 0.0
+
+    def grabVector(self):
+        vectorX = 0.0
+        vectorY = 0.0
+        vectorZ = 0.0
+
+        # Acquire lock, read the values then release lock
+        if self.vectorLock.acquire():
+            vectorX = self.vectorX
+            vectorY = self.vectorY
+            vectorZ = self.vectorZ
+            self.vectorLock.release()
+
+        return (vectorX, vectorY, vectorZ)
+
+    def updateVectors(self, vectorX, vectorY, vectorZ):
+        if self.vectorLock.acquire(False):
+            self.vectorX = vectorX
+            self.vectorY = vectorY
+            self.vectorZ = vectorZ
+            self.vectorLock.release()
+
+    def stream(self, saveData = False):
+        # initialize the bounding box coordinates of the object we are going
+        # to track
+        initBB = None
+
+        # if a video path was not supplied, grab the reference to the web cam
+        print("[INFO] starting video stream...")
+        self.running = True
+        vs = VideoStream(src=0).start()
+        time.sleep(1.0)
+
+        # Initialise variables
+        fps = None
+        preX = 0.0
+        preY = 0.0
+        preZ = 0.0
+
+        vectorX = 0.0
+        vectorY = 0.0
+        vectorZ = 0.0
+
+        vectorDataSet = []
+        startTime = 0
+        # loop over frames from the video stream
+        while self.running:
+            # grab the current frame, then handle if we are using a
+            # VideoStream or VideoCapture object
+            frame = vs.read()
+
+            # check to see if we have reached the end of the stream
+            if frame is None:
+                break
+
+            # resize the frame (so we can process it faster) and grab the
+            # frame dimensions
+            frame = imutils.resize(frame, width=500)
+            (H, W) = frame.shape[:2]
+
+                # check to see if we are currently tracking an object
+            if initBB is not None:
+                # grab the new bounding box coordinates of the object
+                (success, box) = self.tracker.update(frame)
+
+                # check to see if the tracking was a success
+                if success:
+                    (x, y, w, h) = [int(v) for v in box]
+                    cv2.rectangle(frame, (x, y), (x + w, y + h),
+                        (0, 255, 0), 2)
+
+                    vectorX += x + w/2 - preX
+                    vectorY += y + h/2 - preY
+                    vectorZ += 0.0 # Needs work
+                    preX = x + w/2
+                    preY = y + h/2
+
+                    # If unable to acquire the lock, don't update the vectors
+                    if self.vectorLock.acquire(False):
```

Comment:
Missing a `self.vectorLock.release()` to release the lock. Potentially, you may have a lock that never opens